lib/permalink.ts

```diff
@@ -50,7 +50,12 @@ class Permalink {
   }
 
   stringify(data) {
-    return this.rule.replace(rParam, (match, name) => data[name]);
+    return this.rule.replace(rParam, (match, name) => {
+      if (['permalink', 'path'].includes(name)) {
+        throw new Error('Invalid permalink setting!');
```

Comment:
I think this is not the best change, it makes `hexo-util` and `hexo` become coupled. (Although probably no one uses `hexo-util` except `hexo`)

https://github.com/hexojs/hexo-util/blob/ec7fd7cddcaecf8c1c33d34eddd1776bc5e05064/README.md?plain=1#L404-L405

Reply:
I've updated the patch to disallow the case where the parameter is a getter. Please take a look to see if this is better.

Reply:
> I've updated the patch to disallow the case where the parameter is a getter. Please take a look to see if this is better.

This may affect `:tags` and `:categories`, but maybe no one will use these two parameters because they can't be overridden by front-matter

Reply:
Thank you for your work. it looks better than before

Reply:
`:tags` or `:categories` looks like an array